Src/Newtonsoft.Json/Utilities/ConvertUtils.cs

```diff
@@ -577,7 +577,7 @@ private static ConvertResult TryConvertInternal(object initialValue, CultureInfo
                 return ConvertResult.Success;
             }
 #endif
-#if !(DOTNET || PORTABLE40 || PORTABLE)
+#if HAVE_ADO_NET
```

Comment:
Nope, `HAVE_DB_NULL_TYPE` refers to PrimitiveTypeCode.DBNull and replaces `!(DOTNET || PORTABLE`). I'll rename this to `HAVE_DB_NULL_TYPE_CODE` for clarity.